tests/alternative/Network.lua

```diff
@@ -1,155 +1,352 @@
+
+			local roadsLayerCurrDir = gis.Layer{
+				project = proj,
+				name = "roadsCurrDir",
+				file = roadsCurrDir
+			}
+
+			local roadsCs = CellularSpace{project = proj, layer = roadsLayerCurrDir.name, missing = 0}
+
+			local communities = CellularSpace{
+				file = filePath("communities.shp", "gpm")
+			}
+
+			local disconnWithProjectError = function()
+				Network{
+					lines = roadsCs,
+					target = communities,
+					progress = false,
+					inside = function(distance) return distance end,
+					outside = function(distance) return distance end,
+					error = 900
+				}
+			end
+
+			unitTest:assertError(disconnWithProjectError, "The network is disconnected. Layer 'neterror' was automatically created with attribute 'net_id' containing the separated networks.")
+
+			local neterrorLayer = gis.Layer{
+				project = proj,
+				name = "neterror"
+			}
+
+			local attrs = neterrorLayer:attributes()
+			unitTest:assertEquals(attrs[19].name, "net_id")
+
+			proj.file:delete()
+			File(neterrorLayer.file):delete()
+
+			local disconnectedError = function()
+				local roads = CellularSpace{
+					file = roadsCurrDir,
+					missing = 0
+				}
+
+				Network{
+					lines = roads,
+					target = communities,
+					progress = false,
+					inside = function(distance) return distance end,
+					outside = function(distance) return distance * 2 end,
+					error = 900
+				}
+			end
+
+			unitTest:assertError(disconnectedError, "The network is disconnected. Data 'neterror.shp' was automatically created with attribute 'net_id' containing the separated networks.")
+
+			local neterror = CellularSpace{
+				file = "neterror.shp",
+				missing = 0
+			}
+
+			unitTest:assertNotNil(neterror:sample().net_id)
+
+			File(roadsCurrDir):delete()
+			neterror.file:delete()
 		end
 
-		unitTest:assertError(error_func, "Line: '7' does not touch any other line. The minimum distance found was: 843.46359196883.")
+		disconnectedNetworkTest()
+
+		local errorArgumentError = function()
+			local roadsSouth = CellularSpace{
+				file = filePath("test/roads_sirgas2000_south3.shp", "gpm")
+			}
+
+			local ports = CellularSpace{
+				file = filePath("test/porto_alegre_sirgas2000.shp", "gpm"),
+				missing = 0
+			}
 
-		error_func = function()
 			Network{
-				lines = roads,
-				target = communities,
-				inside = function(distance) return distance end,
-				outside = function(distance) return distance * 2 end,
-				error = 900
+				lines = roadsSouth,
+				target = ports,
+				progress = false,
+				error = 400,
+				inside = function(distance)
+					return distance
+				end,
+				outside = function(distance)
+					return distance * 4
+				end
 			}
 		end
 
-		unitTest:assertError(error_func, "The network is disconnected. For example, objects '1' and '2' belong to two separated networks.")
+		unitTest:assertError(errorArgumentError, "Line '47' was added because the value of argument 'error: 400'. Remove the error argument and correct the lines disconnected.")
```

Comment:
Line 47 was added due to the value of argument 'error' (400). Please remove the argument 'error' and fix the disconnected lines.